### PR TITLE
WEO-87 Finalized text and section bar

### PR DESCRIPTION
**Pull request by Awais3214**
Completed two progress bars i.e WOI text bar and WOI section bar. The text bar is square progress widget that contains a text which displays the progress with a numerical value and it is filled with a different color based on the progress so for example if progress is 50 percent, the widget will display 50 which will be half filled. The section bar divides 1 goal into smaller tasks each having a length of user choice with a progress indicator hovering over the bar to indicate the current progress.

### CODE REVIEW

File: widgets/lib/progress_bar/text_bar/text_bar_painter.dart
Comment (saadjavaidalvi-weoveri) [inline widgets/lib/progress_bar/text_bar/text_bar_painter.dart:1]: Please add more comments that confirms which progress bar is related to

**Comment by saadjavaidalvi-weoveri** [comment]
Everything else looks good, but before squash and merge, do look into the conflicts

File: widgets/lib/progress_bar/text_bar/text_bar_painter.dart
Comment (Awais3214) [reply widgets/lib/progress_bar/text_bar/text_bar_painter.dart:1]: I dont understand what is actually required to be done. Could you please elaborate this?

Comment (saadjavaidalvi-weoveri) [reply widgets/lib/progress_bar/text_bar/text_bar_painter.dart:1]: So the class FillPainter is a very generic name which doesn't tell that with whom this relates to, so either adding more documentation comments like this is used in TextProgressBar widget, or update the name to something like FillTextPainter that would explain exactly what this painter is about.

**Pull request merged**